### PR TITLE
[PPP-5350]-XSS Findings For Pentaho-platform-plugin-common-ui

### PR DESCRIPTION
**Pull request by Rangashivani**
[PPP-5350]-XSS Findings For Pentaho-platform-plugin-common-ui

[PPP-5350]: https://hv-eng.atlassian.net/browse/PPP-5350?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

File: impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js [hunk 1]
```diff
@@ -123,7 +124,7 @@ require([
 
     }
 
-    this.numSpan.innerText = value == null ? "" : value;
+    xssUtil.setHtml(this.numSpan, value == null ? "" : value);
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js:127]: `innerText` was fine, here.

File: impl/client/src/main/javascript/web/dojo/pentaho/common/FieldsetPane.js [hunk 2]
```diff
@@ -52,7 +52,7 @@ define([
 
     setTitle: function (/*String*/ title) {
       this.title = title;
-      this.titleNode.innerHTML = DOMPurify.sanitize(title);
+      this.titleNode.innerHTML = xssUtil.sanitizeHtml(title);
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/dojo/pentaho/common/FieldsetPane.js:55]: Why not use `xssUtil.setHtml(..)`?

File: impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js [hunk 3]
```diff
@@ -150,7 +150,7 @@ define([
         domClass.add(btn, "pentaho-button");
         btn.setAttribute( "id", "button"+j);
         // this is filled with the sanitized localized strings, however buttons can be override hence the double check
-        btn.innerHTML = DOMPurify.sanitize(this.buttons[j]);
+        btn.innerHTML = xssUtil.sanitizeHtml(this.buttons[j]);
```
Comment (dcleao) [inline impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js:153]: Why not use `xssUtil.setHtml(..)`?

**Review by dcleao — CHANGES_REQUESTED** [review]

File: impl/client/src/main/javascript/web/dojo/pentaho/common/FieldsetPane.js (hunk 2 above)
Comment (Rangashivani) [reply impl/client/src/main/javascript/web/dojo/pentaho/common/FieldsetPane.js]: Updated

File: impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js (hunk 3 above)
Comment (Rangashivani) [reply impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js]: Updated

File: impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js (hunk 1 above)
Comment (Rangashivani) [reply impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js]: updated

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 17m 8s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5250**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1801/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js (hunk 1 above)
Comment (dcleao) [reply impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js]: No changes are needed on this file. You forgot about the dependency and argument.

Comment (Rangashivani) [reply impl/client/src/main/javascript/web/vizapi/ccc/sample/sampleVisualization.js]: Updated

**Comment by buildguy** [comment]
[comment]: <> (FrogbotReviewComment)

<div align='center'>

[👍 Frogbot scanned this pull request and did not find any new security issues.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 20m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
:ok_hand: All tests passed!
 
Tests run: **5250**, Failures: **0**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/common-ui/job/PR-1801/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Pull request merged**